[reland][quant][graphmode][fx][refactor] Factor out insert_observers_for_model to a separate function (#54733)

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#55307 [reland][quant][graphmode][fx][refactor] Factor out insert_observers_for_model to a separate function (#54733)**

Summary:

Test Plan:

Reviewed By: vkuzo

Differential Revision: [D27567475](https://our.internmc.facebook.com/intern/diff/D27567475)

jerryzh168 committed:
[reland][quant][graphmode][fx][refactor] Factor out insert_observers_for_model to a separate function (#54733)

Summary:

Test Plan:

Reviewed By: vkuzo

[ghstack-poisoned]

mruberry reviewed (approved):
mypy build looks to be passing now

jerryzh168 committed:
Update on "[reland][quant][graphmode][fx][refactor] Factor out insert_observers_for_model to a separate function (#54733)"


Summary:

Test Plan:

Reviewed By: vkuzo

Differential Revision: [D27567475](https://our.internmc.facebook.com/intern/diff/D27567475)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[reland][quant][graphmode][fx][refactor] Factor out insert_observers_for_model to a separate function (#54733)"


Summary:

Test Plan:

Reviewed By: vkuzo

Differential Revision: [D27567475](https://our.internmc.facebook.com/intern/diff/D27567475)

[ghstack-poisoned]
